### PR TITLE
chore: replace logger framework

### PR DESCRIPTION
**Pull request by bonomat**
With this new logger we can easily log into a file and into the console.

Second commit is optional, let me know if you want to keep it or drop it. 

Initial few log files look like this: 

```
flutter: [I] TIME: 2023-10-06T15:41:17.320986 r: Initialized logger
flutter: [I] TIME: 2023-10-06T15:41:17.656526 Coordinator version: 1.3.0
flutter: [I] TIME: 2023-10-06T15:41:17.657005 Client is up to date: 1.3.0
flutter: [I] TIME: 2023-10-06T15:41:19.653350 App data will be stored in: /Users/bonomat/Library/Developer/CoreSimulator/Devices/BBD86D27-7B1E-4237-BBDB-34FF9F250C78/data/Containers/Data/Application/E8A20B9D-0E4D-4568-A60C-5FD5FA6337D4/Documents
flutter: [I] TIME: 2023-10-06T15:41:19.653961 Seed data will be stored in: /Users/bonomat/Library/Developer/CoreSimulator/Devices/BBD86D27-7B1E-4237-BBDB-34FF9F250C78/data/Containers/Data/Application/E8A20B9D-0E4D-4568-A60C-5FD5FA6337D4/Library/Application Support
flutter: [D] TIME: 2023-10-06T15:41:19.661211 r: Parsing config from flutter config: Config { coordinator_pubkey: "02dd6abec97f9a748bf76ad502b004ce05d1b2d1f43a9e76bd7d85e767ffb022c9", esplora_endpoint: "http://127.0.0.1:3000", host: "127.0.0.1", p2p_port: 9045, http_port: 8000, network: "regtest", oracle_endpoint: "http://127.0.0.1:8081", oracle_pubkey: "16f88cf7d21e6c0f46bcbc983a4e3b19726c6c98858cc31c83551a88fde171c0", health_check_interval_secs: 2 }
flutter: [D] TIME: 2023-10-06T15:41:19.685110 r: Database migration run - db initialized
```

Log file for the app in the simulator will be located in  (depending on your device id)

`~/Library/Developer/CoreSimulator/Devices/BBD86D27-7B1E-4237-BBDB-34FF9F250C78/data/Containers/Data/Application/A654B954-9614-40F7-9984-1C253FF929E4/Library/Application Support/app_log.log`

@klochowicz : this solves the problem of losing logs for the app as the logs are stored in a file right away.

fixes: https://github.com/get10101/10101/issues/1202

### CODE REVIEW

File: mobile/lib/common/settings_screen.dart [hunk 1]
```diff
@@ -202,7 +203,11 @@ class _SettingsScreenState extends State<SettingsScreen> {
         ),
         ElevatedButton(
             onPressed: () async {
-              var file = await FLog.exportLogs();
+              // TODO:
+              logger.toString();
+
+              // TODO: fix me
```
Comment (holzeis) [inline mobile/lib/common/settings_screen.dart:209]: Does exporting the logs still work with that framework?

Comment (klochowicz) [reply mobile/lib/common/settings_screen.dart:209]: I tried to tackle it once upon a time, and couldn't get this to work easily, so instead went for a quickfix during Dart3 upgrade: #1092 .

if I remember correctly there was no API, but we should be able to read the stream into memory and export it

**Comment by klochowicz** [comment]
> @klochowicz : this solves the problem of losing logs for the app as the logs are stored in a file right away.

Thanks for tackling this, I've reopened #1092 as this was the suggestion in that issue.

one addition though: for this to be useful in development, we should tackle #1202 as well. It's hard to piece this information together otherwise.

File: mobile/lib/common/settings_screen.dart (hunk 1 above)
Comment (bonomat) [reply mobile/lib/common/settings_screen.dart:209]: Sorry, yes it works. The fix me is outdated.

**Comment by bonomat** [comment]
> one addition though: for this to be useful in development, we should tackle https://github.com/get10101/10101/issues/1202 as well. It's hard to piece this information together otherwise.

You make it sound like this PR would not be useful unless we do https://github.com/get10101/10101/issues/1202 😅

Anyway, I did it :)

File: mobile/lib/common/settings_screen.dart (hunk 1 above)
Comment (holzeis) [reply mobile/lib/common/settings_screen.dart:209]: Can we remove it then? 🙂

File: mobile/lib/logger.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import 'package:logger/logger.dart';
+
+Logger get logger => AppLogger.instance;
+
+class AppLogger {
+  static final Logger _logger = Logger(
+      output: HybridOutput(),
+      printer: SimpleUTCPrinter(
+          // Colorful log messages
+          colors: false,
```
Comment (holzeis) [inline mobile/lib/logger.dart:12]: 🙃 why not with colors :)

File: mobile/lib/main.dart [hunk 3]
```diff
@@ -64,17 +63,16 @@ import 'package:package_info_plus/package_info_plus.dart';
...
 
 void main() async {
   WidgetsBinding widgetsBinding = WidgetsFlutterBinding.ensureInitialized();
   FlutterNativeSplash.preserve(widgetsBinding: widgetsBinding);
 
-  setupFlutterLogs();
-
   await initFirebase();
-  FLog.info(text: "Initialised firebase!");
+  logger.i("Logger is working!");
```
Comment (holzeis) [inline mobile/lib/main.dart:75]: This can probably be removed again.

**Review by holzeis — APPROVED** [review]
Nice 👍

**Comment by holzeis** [comment]
> Second commit is optional, let me know if you want to keep it or drop it.

I don't care to much for this prefix either way I am ok.

File: mobile/lib/logger.dart (hunk 2 above)
Comment (bonomat) [reply mobile/lib/logger.dart:12]: It makes the log text longer and you won't see it anyways in the file afterwards.

**Pull request merged**